pkg/components/images_test.go

```diff
@@ -61,20 +63,22 @@ var _ = Describe("test GetReference", func() {
 			Entry("a tigera image correctly", ComponentTigeraNode, "tigera/cnx-node"),
 			Entry("an ECK image correctly", ComponentElasticsearchOperator, "tigera/eck-operator"),
 			Entry("an operator init image correctly", ComponentOperatorInit, "tigera/operator"),
-			Entry("a CSR init image correctly", ComponentCSRInitContainer, "tigera/key-cert-provisioner"),
+			Entry("a CSR init image correctly", ComponentCSRInitContainer, "calico/key-cert-provisioner"),
+			Entry("a CSR init image correctly", ComponentCSRInitContainerPrivate, "tigera/key-cert-provisioner"),
 		)
 	})
 
 	Context("image prefix override", func() {
 		DescribeTable("should render",
 			func(c component, image string) {
-				Expect(GetReference(c, "quay.io/", "", "pref", nil)).To(Equal(fmt.Sprintf("quay.io/%s:%s", image, c.Version)))
+				Expect(GetReference(c, "quay.io/", "", "pref/", nil)).To(Equal(fmt.Sprintf("quay.io/%s:%s", image, c.Version)))
```

Comment:
This might work but it doesn't match the expectation for how the prefix field is expected to be used. The prefix is expected to be a string added to the start of the image name. This test change suggests is is for setting the path.
The image format is documented as `<registry><imagePath>/<imagePrefix><imageName>:<image-tag>`

Reply:
Tbh I found it unpleasant to read the code easily, which motivated the change for me. I didn't think it broke the contract with the API, I just picked something that I preferred reading.
That said, I will revert it.

Reply:
I'd suggest `prefix-` if you think having a separator is more readable.